File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var specialArray = function (nums) {
  for( let i=0 ;i<= nums.length ; i++ )
  {
    let n = 0;
    for( let j =0 ;j< nums.length ; j++ )
    {
      if( nums[ j ] >= i  )
      n ++ ;

    }
    if( n == i )
    return i;
  }
  return -1;
};

File: 1608-special-array-with-x-elements-greater-than-or-equal-x/1608-special-array-with-x-elements-greater-than-or-equal-x.cpp
class Solution {
public:


int specialArray(vector<int>& nums)
{
    int l =0 , r= nums.size();
    while( l<=r)
    {
        int mid = ( l+r)/2;
        int n=0 ;
        for( int i=0 ;i< nums.size() ;i++ )
        {
            if(nums[i]>=mid)
                n++;
        }
        if( n == mid )
            return mid;
        else if( n>mid )
        {
                        l= mid+1;

        }
        else
        r= mid-1;
    }
    return -1;
}
};